### PR TITLE
BUGFIX: NodeTypesLoader respect package overrides

### PR DESCRIPTION
**Pull request by gjwnc**
**What I did**

The NodeTypesLoader introduced in FLOW 7.2 does not allow to override NodeTypes in the newly introduced `NodeTypes` directory. This is because, the `Configuration` directory is loaded in a separate loop after the `NodeTypes` directory.  

For example, the NodeType `Neos.Neos:ContentCollection` cannot be overriden in the `NodeTypes` directory, because it is loaded in `Neos.Neos/Configuration`. This means, everything defined in the `NodeTypes` directory of a custom package is overriden by the `Neos.Neos/Configuration` NodeType.

**How I did it**

This PR uses a single `$packages` loop to first load from the `NodeTypes` directory, directly followed by the `Configuration` for a given `$package`.

**How to verify it**

Try to modify `Neos.Neos:ContentCollection` like this
```
'Neos.Neos:ContentCollection':
  constraints:
    nodeTypes:
      '*': false
```
in a `NodeTypes` directory.

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

### CODE REVIEW

**Comment by mhsdesign** [comment]
solves: https://github.com/neos/neos-development-collection/issues/3449

**Review by dlubitz — APPROVED** [review]
Looks good by reading!

**Review by markuspfeifenberger — APPROVED** [review]
Work for me

**Comment by bwaidelich** [comment]
Nice, thanks for taking care!
Is this a potentially breaking change now that we released a version that handles the inheritance differently or can we assume that nobody expects the current behavior?
In any case we should cover this with a test.. Maybe I can help with that

**Comment by gjwnc** [comment]
>  Is this a potentially breaking change now that we released a version that handles the inheritance differently or can we assume that nobody expects the current behavior?

I think the current 7.2 behavior is just a bug, because it breaks the expected loading order of packges as [documented](https://flowframework.readthedocs.io/en/stable/TheDefinitiveGuide/PartIII/PackageManagement.html#loading-order). The `NodeTypes` folder is just a feature which is part of a package. If we load the configuration of packages I think we should load everything of this single package in one go - except the FLOW__CONTEXT related Configuration which is loaded at the very end of NodeTypes loading (as it was already before 7.2 because this is highly specialized for a project).

>  In any case we should cover this with a test.. Maybe I can help with that

Would be great if you could provide a test. I don't really know how to provide a good test for the NodeTypesLoader because it operates on real files and directory hierarchies. I don't know where to put this for a unit test?  
I've added you as a maintainer to the [repo](https://github.com/webandco/neos-development-collection/tree/bugfix/nodetype-overrides).

**Comment by mhsdesign** [comment]
you could mock the file system with vfsstream @gjwnc

**Comment by bwaidelich** [comment]
> you could mock the file system with vfsstream

Yes exactly, I'm on it already. Follow-up incoming

**Comment by bwaidelich** [comment]
I added some unit tests now covering the override logic..
Unfortunately they run through with the old _and_ the new code currently 🤔 
I'll have to add some more cases I guess

**Comment by bwaidelich** [comment]
Just to make sure: @dlubitz & @markuspfeifenberger can you confirm that this change actually fixes the issue described in #3449 ?

**Review by bwaidelich — APPROVED** [review]
OK, this took much longer than I had hoped but `vfsStream` didn't work out in the end because it lacks support for `glob` (see https://github.com/bovigo/vfsStream/issues/2) which we use in our `YamlSource`.
So I replaced everything with actual fixture files and can confirm that the bug is fixed and that it doesn't introduce any unplanned side effects \o/

Will merge if checks are green!

**Comment by bwaidelich** [comment]
phew, finally..

**Comment by mhsdesign** [comment]
@bwaidelich the YamlSource could use instead of glob the RecursiveDirectoryIterator or something - this is supported by vfsstream. but i guess this isnt that important...

**Pull request merged**
